### DISCUSSION TITLE
No matching distribution found for wheel~=0.42

### DISCUSSION THREAD

**Discussion by Askmecoding** [projen/projen#3922] [Q&A]
I am getting this issue while running npx projen build. I have this in my projen file: cdkVersion: '2.164.1' and jsiiVersion: '5.5.4',

```
[jsii-pacmak] [INFO] Command "/var/folders/fs/62rfen/T/npm-pack69AJcP/.env-7rBt0n/bin/python -m pip install --no-input -r /ws/templates/cdkconstructs/node_modules/jsii-pacmak/lib/targets/python/requirements-dev.txt" failed with Command (/var/folders/fs/62r0mmwd2xq6kp6l3yg7h_sh0000gn/T/npm-pack69AJcP/.env-7rBt0n/bin/python -m pip install --no-input -r /ws/templates/cdkconstructs/node_modules/jsii-pacmak/lib/targets/python/requirements-dev.txt) failed with status 1:
#STDERR> ERROR: Could not find a version that satisfies the requirement wheel~=0.42 (from versions: none)
#STDERR> ERROR: No matching distribution found for wheel~=0.42
#STDERR>
```

I am using custom repositories. For some reason it was going to different repo where it did not have access. I was able to update repo url on my .pypirc and fix the issue.

Is there a way to define custom repositories (instead of maven central or pther public repositories) for dependency resolution for CDK constructs library for python, maven and npm?

**Comment by mrgrain** [comment]
wheel 0.42.0 exists though.
https://pypi.org/project/wheel/0.42.0/

You sure this failure isn't transient?
Are you using any custom registries?

**Comment by mrgrain** [comment]
> Is there a way to define custom repositories (instead of maven central or pther public repositories) for dependency resolution for CDK constructs library for python, maven and npm?

This isn't straight forward in projen at the moment, especially if you need to define this for multiple languages. Here are some starting points:

NPM:
- https://projen.io/docs/api/javascript?_highlight=registry#projen.javascript.NpmConfigOptions

Python:
- Poetry: https://projen.io/docs/api/python?_highlight=poetr#projen.python.PoetryPyprojectOptions.property.source
- Or set env variables based on your package manager:
  - https://docs.readthedocs.io/en/stable/guides/private-python-packages.html#from-a-repository-manager-other-than-pypi
  - https://python-poetry.org/docs/repositories/#publishing-to-a-private-repository

Maven: 
- https://maven.apache.org/guides/mini/guide-multiple-repositories.html
- In projen use [addRepository](https://projen.io/docs/api/java#projen.java.Pom.addRepository)
